src/server/routes/api/document/index.js

```diff
@@ -1295,6 +1307,52 @@ http.post('/', function( req, res, next ){
     if( id === DEMO_ID ) await deleteTableRows( API_KEY, secret );
     return await createDoc({ docDb, eleDb, id, secret, provided });
   };
+  const addEls = doc => tryPromise( () => doc.fromJson( { elements } ) ).then( () => doc );
+  const handleLayout = doc => {
+    if ( performLayout ) {
+      return tryPromise( () => doc.applyLayout() ).then( () => doc );
+    }
+
+    return doc;
+  };
+  const handleSubmission = doc => {
+    if ( submit ) {
+      return tryPromise( () => doc.submit() ).then( () => doc );
+    }
+
+    return doc;
+  };
+  const handleElGroundings = doc => {
+    const perform = () => {
+      let entities = doc.entities();
+      let intns = doc.interactions();
+
+      let entityPromises = entities.map( entity => {
+        let name = entity.name();
+        let opts = {
+          limit: 1,
+          offset: 0,
+          name
+        };
+
+        return searchGrounding( opts ).then( res => {
+          if( res && res.length > 0 ) {
+            return entity.associate( res[0] );
+          }
+        } );
+      } );
+
+      let intnPromises = intns.map( intn => intn.associate( 'interaction' ) );
```

Comment:
Are all the interactions of the general 'interaction' type, or do we use specific types like 'binding'?

Reply:
@maxkfranz you are right it may possibly be other vlaues and actually it is already specified in ``association`` and ``tyoe`` fields (I always put the same value to both) in interaction json. Therefore, I totally removed the related code now.

Reply:
Great